generators/base-workspaces/internal/docker-prompts.ts

```diff
@@ -73,13 +68,13 @@ async function askForApplicationType() {
 
   const props = await this.prompt(prompts, this.config);
   this.deploymentApplicationType = props.deploymentApplicationType;
-}
+});
 
 /**
  * Ask For Gateway Type
  */
-async function askForGatewayType() {
-  if (!this.options.askAnswered && (this.regenerate || this.config.existed)) return;
+export const askForGatewayType = asPromptingWorkspaceTask(async function askForGatewayType(this: BaseWorkspacesGenerator, { control }) {
+  if (!this.options.askAnswered && (this.regenerate || control.existingProject)) return;
```

Comment:
We should try to drop `this.regenerate` and use the function shouldAskForPrompts

Reply:
This can be done by tweaking existingProject.

Reply:
Will do it later, thanks for the advice